fix(c): Remove duplicate code generation for UUID path parameters

BiekerUdan opened the pull request:
The C generator was generating duplicate path parameter handling code when a parameter had format: uuid. This occurred because UUID parameters have both isString=true and isUuid=true flags set, causing both the {{#isString}} and {{#isUuid}} template blocks to execute.

The duplicate {{#isUuid}} block has been removed since it generated identical code to the {{#isString}} block. This makes UUID parameters consistent with email parameters, which already work this way (isEmail=true and isString=true, but only use the {{#isString}} block).

The generated code now compiles successfully without duplicate variable declarations.

Fixes #17448

@zhemant @ityuhui @michelealbano @eafer

<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package || exit
  ./bin/generate-samples.sh ./bin/configs/*.yaml || exit
  ./bin/utils/export_docs_generators.sh || exit
  ``` 
  (For Windows users, please run the script in [WSL](https://learn.microsoft.com/en-us/windows/wsl/install))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR solves a reported issue, reference it using [GitHub's linking syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) (e.g., having `"fixes #123"` present in the PR description)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

wing328 commented:
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

eafer commented:
Do you mind adding a test for your change to the `petstore.yaml` for C? So that it doesn't break again in the future.

BiekerUdan commented:
I fixed the attribution issue. And added '/pet/byUuid/{uuid}' to petstore.yaml and generated the samples.

This causes builds of the samples to fail before this PR is applied, and to pass after it is applied.  I did not add any unit tests specifically for this, is that ok?

eafer commented:
Yes, that's what I meant: just put something in place so that the sample build fails if we ever revert this by mistake. Thanks! Your changes look good to me, but I don't think all of the samples got refreshed for some reason.
